java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -24,18 +24,99 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 import java.util.function.Consumer;
 
+import static ai.rapids.cudf.HostColumnVector.OFFSET_SIZE;
+
 /**
  * This class represents the immutable vector of data.  This class holds
  * references to device(GPU) memory and is reference counted to know when to release it.  Call
  * close to decrement the reference count when you are done with the column, and call incRefCount
  * to increment the reference count.
  */
-public final class ColumnVector implements AutoCloseable, BinaryOperable {
+public final class ColumnVector implements AutoCloseable, BinaryOperable, ColumnViewPointerAccess<BaseDeviceMemoryBuffer> {
   private static final Logger log = LoggerFactory.getLogger(ColumnVector.class);
 
+  public class DeviceColumnViewPointerAccess implements ColumnViewPointerAccess<BaseDeviceMemoryBuffer> {
+
+    protected long viewHandle;
+
+    public DeviceColumnViewPointerAccess(long viewHandle) {
+      this.viewHandle = viewHandle;
+    }
+
+    @Override
+    public long getColumnView() {
```

Comment:
I am wondering if this method should be renamed to something that conveys what its returning. may be `getColumnViewAddress`??

Reply:
That wont be like `getNativeView` anymore then. Let me know what u think of that.

Reply:
That is a private method. I am mostly concerned about the public API calling it a view when its a long.

Reply:
u got it! I will rename this.